app/components/modal/body_component.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class Modal::BodyComponent < ViewComponent::Base
+  def initialize(text: nil, klass: nil, render_check: true)
+    @text = text
+    @render_check = render_check
+    @class = klass
+  end
+
+  def body_content
+    return content if content.present?
+
+    Array.wrap(@text).map do |text|
+      content_tag :p, text
+    end.join.html_safe
+  end
```

Comment:
This seems pretty restrictive to what can go in to the body. Why not just insert what the content is and do something like this when rendering...?
```
<% component.with_body do %>
  <p>
    This topic and its related questions...
  </p>
  <p>
    This will not affect case contacts that...
  </p>
<% end %>
```

Reply:
You can do that, currently if you provide a block it overrides and text is optional. 

It's just if you provide text as a string or array of strings. It gets wrapped in ptags. 

https://github.com/rubyforgood/casa/blob/897d5f79737f894c32dc0e28a0570daf8f833f6d/spec/components/modal/body_component_spec.rb#L20